Ft login 160577439

michael-basweti opened the pull request:
## What does this PR do?
* implements login
## Description of Task to be completed?
* users should be able to login and access services after having been properly authenticated
## How should this be manually tested?
* clone [this](https://github.com/andela/ah-shakas-frontend.git) repo.
* run command `git checkout ft-login-160577439`.
* run `yarn install` to install requirements
* run `yarn start` to run project
* navigate to [http://localhost:3000/](http://localhost:3000/) locally to access the system locally or [http://10.10.162.103:3000/](http://10.10.162.103:3000/)  to access it online.
## Any background context you want to provide?
* you should be registered to the system to be able to login
## What are the relevant pivotal tracker stories?
* [#160577439](https://www.pivotaltracker.com/n/projects/2198533/stories/160577439)
## Packages installed
```
"axios": "^0.18.0",
    "enzyme": "^3.7.0",
    "enzyme-adapter-react-16": "^1.6.0",
    "eslint-config-airbnb": "^17.1.0",
    "history": "^4.7.2",
    "jest": "^23.6.0",
    "node-sass": "^4.9.4",
    "prop-types": "^15.6.2",
    "react": "^16.6.0",
    "react-dom": "^16.6.0",
    "react-materialize": "^2.4.8",
    "react-redux": "^5.1.0",
    "react-router-dom": "^4.3.1",
    "react-scripts": "2.0.5",
    "react-test-renderer": "^16.6.0",
    "redux": "^4.0.1",
    "redux-devtools": "^3.4.1",
    "redux-devtools-extension": "^2.13.5",
    "redux-logger": "^3.0.6",
    "redux-promise-middleware": "^5.1.1",
    "redux-thunk": "^2.3.0"
```

## Deployment note
N/A

## Related PRs branch | PR (branch_name) | (pr_link)
Branch | PR
------ | ------
ft-signup-160577437 | [signup](https://github.com/andela/ah-shakas-frontend/tree/ft-signup-160577437)

## Todos
- [] Raise PR
- [x] Test
- [ ] Documentation


## Screenshots 
# login page
<img width="1280" alt="screenshot 2018-11-08 at 15 13 31" src="https://user-images.githubusercontent.com/23398223/48199226-91fc3c00-e36c-11e8-9b4d-d0d2f437c5c5.png">
#login with blank email and password
<img width="1276" alt="screenshot 2018-11-08 at 15 13 54" src="https://user-images.githubusercontent.com/23398223/48199266-ab9d8380-e36c-11e8-8fe1-d5489fe52c83.png">

michael-basweti committed:
start feature

michael-basweti committed:
feat(kafka): implement login

- add actions
- added reducers
- Component for login

[WIP #160577439]

michael-basweti committed:
feat(login): implement login

- add actions
- added reducers
- Component for login
-add validations

[WIP #160577439]

Arthur236 commented on src/App.scss:
```diff
@@ -30,3 +30,12 @@
+
+}
+
+.MYCHECK{
```
Use lower case letters for your classes

Arthur236 commented:
@michael-basweti work on fixing the linting errors

Arthur236 commented:
Separate you container logic and component logic as discussed today during the code review

michael-basweti committed:
add tests

michael-basweti committed:
feat(login): implement login

- add actions
- added reducers
- Component for login
-add validations
-add tests
-separate my container and component

[WIP #160577439]

michael-basweti committed:
Fix Hound errors

michael-basweti committed:
fix failing tests for actions

michael-basweti committed:
fix errors in tests

Arthur236 commented on src/__tests__/loginReducers/login.reducer.test.js:
```diff
@@ -0,0 +1,33 @@
+    action.type = `${LOGIN}_PENDING`;
+    expect(loginReducer(initialState.login, action).onPending).toEqual(true);
+  });
+  it('should handle USER_LOAD_REQUEST', () => {
```
Have a blank line to separate each of your `it` blocks.
Apply this in all your tests

Arthur236 commented on src/components/Dashboard/dashboard.jsx:
```diff
@@ -0,0 +1,33 @@
+import { Link } from 'react-router-dom';
+import { Navbar } from 'react-materialize';
+
+class DashboardComponent extends Component {
```
Rename the file to `Dashboard.jsx`

Arthur236 commented on src/components/Dashboard/dashboard.jsx:
```diff
@@ -0,0 +1,33 @@
+    }
+  	const Logout = () => {
+      if (userToken) {
+      	localStorage.removeItem('userToken');
```
Fix this violation

Arthur236 commented on src/constants/login.type.js:
```diff
@@ -0,0 +1 @@
+export const LOGIN = 'LOGIN';
```
Put the constant in the right file as discussed

Arthur236 commented on src/reducers/initialState.js:
```diff
@@ -3,4 +3,8 @@ export default {
+  login: {
+    onPending: false,
+    onFulfilled: false,
+  },
```
I feel we should also have an `onRejected` state here

michael-basweti committed:
feat(login): implement login

- work on feedback
- refactor code to accomodate modals for sign in and sign up

[Finishes #160577439]

Arthur236 commented:
Rename `src/components/Dashboard/dashboard.jsx` to `src/components/Dashboard/Dashboard.jsx`

Arthur236 commented on src/App.scss:
```diff
@@ -30,3 +30,33 @@
+
+.btn-secondary{
+  padding: 0.5rem 2rem;
+  border: 1px teal solid;
```
Use hexadecimal instead of literals for colors.
Apply this everywhere you've used literals

Arthur236 commented on src/components/Auth/LoginComponent.jsx:
```diff
@@ -1,33 +1,89 @@
+      if (userToken) {
+        localStorage.setItem('userToken', userToken);
+        // redirect.history.push('/dashboard');
+        window.location.replace('/dashboard');
```
Why use window instead of the history prop?

Arthur236 commented on src/components/Dashboard/dashboard.jsx:
```diff
@@ -0,0 +1,33 @@
+  render() {
+    const userToken = localStorage.getItem('userToken');
+    if (!userToken) {
+      window.location.replace('/');
```
Use history prop instead of window

Arthur236 commented:
Delete this file `src/constants/login.type.js`. You dont need it now that you've defined your constant in the index file

Arthur236 commented:
<img width="459" alt="screenshot 2018-11-13 at 08 32 31" src="https://user-images.githubusercontent.com/31339738/48393243-d6e2f280-e71f-11e8-9657-3681ace3e5de.png">

Style the buttons as links so that they are consistent with the Home link

Arthur236 commented:
<img width="710" alt="screenshot 2018-11-13 at 08 32 59" src="https://user-images.githubusercontent.com/31339738/48393273-ef530d00-e71f-11e8-8d21-a7f8856010c4.png">

<img width="740" alt="screenshot 2018-11-13 at 08 33 13" src="https://user-images.githubusercontent.com/31339738/48393275-f1b56700-e71f-11e8-9683-f47704c98bff.png">

* These two modals do not have a consistent design. The titles look different and the inputs look a bit different as well. Fix the Sigin modal to look like the Signup.

* Reduce the overly large shadow on the Submit and Sigup buttons

* Change the Submit text to Signin

* There should not be any padding/margin around the images to the left. Based on the designs that were shared yesterday, the modal shold have a similar design to this: 

<img width="759" alt="screenshot 2018-11-13 at 08 46 30" src="https://user-images.githubusercontent.com/31339738/48393440-a3ed2e80-e720-11e8-9d52-2319f621d56a.png">

<img width="760" alt="screenshot 2018-11-13 at 08 45 58" src="https://user-images.githubusercontent.com/31339738/48393441-a485c500-e720-11e8-8804-b269a1d55406.png">

* Notice how the above modals are the same size as well

michael-basweti committed:
feat(login): implement login

- work on feedback

[Finishes #160577439]

michael-basweti committed:
feat(login): implement login

- refactor failing tests

[Finishes #160577439]

michael-basweti committed:
feat(login): implement login

- refactor failing tests

[Finishes #160577439]

michael-basweti committed:
feat(login): implement login

- fix hound linting error

[Finishes #160577439]

Arthur236 commented:
Have you looked at the hound violations?

Arthur236 commented:
<img width="497" alt="screenshot 2018-11-14 at 13 50 40" src="https://user-images.githubusercontent.com/31339738/48478106-64077380-e814-11e8-8f6c-f33a123159a5.png">

There's still a margin at the bottom of the picture. Kindly fix it

Arthur236 commented on src/App.scss:
```diff
@@ -30,3 +32,63 @@
+  background: #ffffff;
+  color: $primary-color;
+  transition: background 500ms ease;
+
```
Remove the blank line

michael-basweti committed:
remove margin

michael-basweti committed:
fix hound errors

michael-basweti committed:
fix hound errors

michael-basweti committed:
fix hound errors

michael-basweti committed:
fix hound errors

Arthur236 commented on src/index.scss:
```diff
@@ -24,14 +24,13 @@ body {
   padding-top: 20px;
-  width: 550px;
+  width: 350px;
+
```
Remove the blank line

michael-basweti committed:
add sass_lint

michael-basweti committed:
rebase

michael-basweti committed:
fix hound

michael-basweti committed:
fix hound

michael-basweti committed:
fix hound

michael-basweti committed:
fix hound

michael-basweti committed:
hound

michael-basweti committed:
sass-lint

michael-basweti committed:
sass

michael-basweti committed:
hound

michael-basweti committed:
hound

michael-basweti committed:
hound

michael-basweti committed:
hound

michael-basweti committed:
fix hound

Arthur236 commented on .sass-lint.yml:
```diff
@@ -0,0 +1,27 @@
+  include: '**/*.s+(a|c)ss'
+rules:
+  # Disallows
+  indentation:
```
Why did you add this rule?
We should be using 2 spaces for indentation

Arthur236 commented:
This file should not be json: `.stylelintrc.json`. It should be `.stylelintrc`

Arthur236 commented on package.json:
```diff
@@ -66,14 +71,7 @@
-        "git add"
-      ]
+      "*{.js,jsx}": "yarn eslint-check"
     }
```
Why did you remove the sass linting?
Kindly return it

michael-basweti committed:
hound

michael-basweti committed:
test hound

michael-basweti committed:
fix hound  errors

Arthur236 reviewed (approved):
Looks good 👍
